juju/juju
# feat(domain): add machine reboot watcher

This commit introduces functionality for watching machine reboot events.
    - Before this commit, there was no mechanism to watch for machine reboot events.
    - After this commit, the system can monitor machine reboots, generate relevant change events
      including both parent and children if a parent machine has been requested to reboot.





## Checklist




## QA steps



Run unit tests.

`TEST_PACKAGES="./domain/machine/... -count=1 -race -check.vv" make run-go-tests`

## Documentation changes

None


## Links



**Jira card:** [JUJU-6304](https://warthogs.atlassian.net/browse/JUJU-6304)
